src/widgets/HtmlText.cpp

```diff
@@ -209,22 +209,23 @@ void HtmlText::LayoutCell()
         }
         else // ! HT_NOWRAP
         {
-            // Increase cell size
-            while (pointSize < m_maxFontSize
-                   && (m_cell->GetHeight() < height
-                       || m_cell->GetWidth() < width))
-            {
-                Parse(label, ++pointSize, faceName);
-                m_cell->Layout(width);
-            }
-            // Decrease cell size
-            while (pointSize > m_minFontSize
-                   && (m_cell->GetHeight() > height
-                       || m_cell->GetWidth() > width))
-            {
-                Parse(label, --pointSize, faceName);
+            // Binary search to find the highest point size where the text still fits.
+            int lowerBound = m_minFontSize;
+            int upperBound = m_maxFontSize;
+            while (lowerBound != upperBound) {
+                pointSize = (lowerBound + upperBound + 1) / 2;
+                Parse(label, pointSize, faceName);
                 m_cell->Layout(width);
+                if (m_cell->GetHeight() < height) {

```

Comment:
I take it the `GetWidth()` check from the above code was redundant? This is pretty old code so I'm honestly not sure.

Reply:
I believe so, but not certain. This if block is in the case that the HT_NOWRAP flag is not set, which I took to mean that wrapping is enabled. If so, then there's no need to check that the text fits the width of the container, because word wrapping would ensure that already.

Maybe there's a pathological case where you have a very long clue with no spaces that can't be wrapped? If so, I could add the width check back.

Reply:
If this doesn't obviously break I'm not worried about it. I was probably overly cautious with the original code.